pilot/pkg/serviceregistry/kube/controller/endpointcontroller.go

```diff
@@ -111,3 +116,47 @@ func updateEDS(c *Controller, epc kubeEndpointsController, ep interface{}, event
 		}
 	}
 }
+
+// reSyncEndpoints syncs endpoints that were not processed because of unavailibility of pod
+// when endpoint event came. Try to reprocess it to see if pod is available or endpoint it self
+// might have been deleted.
+func (e *kubeEndpoints) reSyncEndpoints(stopCh <-chan struct{}) {
+	ticker := time.NewTicker(1 * time.Minute) // TODO: Make this resync period configurable.
```

Comment:
Trying to reason about the effect of the resync period.  I assume it is ok for a new endpoint to be delayed putting into service for up to a minute, but it is undesirable for a dead endpoint be delayed for a such long time till it is removed.

When the endpoint object is being updated frequently as part of the rollout for example, we don't rely on this to keep the list of endpoint up-to-date, but it may impact the last batch of the pods.

Based on my limited understanding of the code, I think in the case of pod deletion but endpoint update comes before pod cache update, it has no negative effect, because the deleted endpoint IP will not be included anyway so we avoid the undesirable effect.  In the case of pod addition but endpoint update comes before pod cache update, it may take up to this resync period to bring it into effect.

Is my understanding correct?

Reply:
Yes that is correct. Any suggestions/improvement here if we decide to go via this path?

Reply:
The only question I have left is, what is preventing us from running this resync as often as possible.  Is it too expensive?  Does not appear so.  In most cases it may have an empty resync set.

Reply:
I do not think it is expensive - in most cases it just comes out immediately. I just chose 1 min randomly. What do you think would be a reasonable resync period?

Reply:
In general, is there a target for control plane/data plane sync latency?  I suggest we make this value to be within that.